src/components/AdminAllBusinesses/AdminAllBusinesses.jsx

```diff
@@ -8,6 +10,21 @@ const AdminAllBusinesses = () => {
   const [pendingFlag, setPendingFlag] = useState(false);
   const [formFlag, setFormFlag] = useState(false);
   const [formItem, setFormItem] = useState({});
+  const [isAdminUser, setIsAdminUser] = useState(false);
+
+  const { isAdmin } = useAuth();
+  const navigate = useNavigate();
+
+  useEffect(() => {
+     const checkIsAdmin = async () => {
+      if (!(await isAdmin())) {
+        navigate('/BusinessDashboard');
+      } else {
+        setIsAdminUser(true);
+      }
+    };
+    checkIsAdmin();
+  }, []);
```

Comment:
⚠️ **[eslint]** <[react-hooks/exhaustive-deps](https://github.com/facebook/react/issues/14920)> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>React Hook useEffect has missing dependencies: 'isAdmin' and 'navigate'. Either include them or remove the dependency array.